ios/PurchasesHybridCommon/PurchasesHybridCommonTests/EntitlementInfo+HybridAdditionsTests.swift

```diff
@@ -0,0 +1,166 @@
+//
+//  EntitlementInfo+HybridAdditionsTests.swift
+//  PurchasesHybridCommonTests
+//
+//  Created by Will Taylor on 5/24/22.
+//  Copyright © 2022 RevenueCat. All rights reserved.
+//
+
+import Quick
+import Nimble
+@testable import PurchasesHybridCommon
+@testable import RevenueCat
+
+class EntitlementInfoHybridAdditionsTests: QuickSpec {
+
+    private static let expiresDate = "9999-01-01T00:00:00Z"
+    private static let purchaseDate = "1999-07-26T23:30:41Z"
+    private let mockEntitlementData = [
+        "expires_date": EntitlementInfoHybridAdditionsTests.expiresDate,
+        "product_identifier": "pro",
+        "purchase_date": EntitlementInfoHybridAdditionsTests.purchaseDate
+    ]
+
+    override func spec() {
+        describe("rc_dictionary") {
+            context("store") {
+                it("equals APP_STORE when the store is app store") {
+                    let expectedDictionaryValue = "APP_STORE"
+                    guard let mockEntitlementInfo = EntitlementInfo(
+                        entitlementId: "pro",
+                        entitlementData: self.mockEntitlementData,
+                        productData: self.mockProductData(withStore: "app_store"),
+                        requestDate: nil
+                    ) else {
+                        fail("Could not create mock EntitlementInfo object.")
+                        return
+                    }
+
+                    let dictionary = mockEntitlementInfo.dictionary
+                    expect(dictionary["store"] as? String).to(
+                        equal(expectedDictionaryValue),
+                        description: "Expected \(String(describing: dictionary["store"] as? String)) to become \(expectedDictionaryValue)."
+                    )
+                }
+                it("equals MAC APP STORE when the store is mac app store") {
+                    let expectedDictionaryValue = "MAC_APP_STORE"
+                    guard let mockEntitlementInfo = EntitlementInfo(
+                        entitlementId: "pro",
+                        entitlementData: self.mockEntitlementData,
+                        productData: self.mockProductData(withStore: "mac_app_store"),
+                        requestDate: nil
+                    ) else {
+                        fail("Could not create mock EntitlementInfo object.")
+                        return
+                    }
+
+                    let dictionary = mockEntitlementInfo.dictionary
+                    expect(dictionary["store"] as? String).to(
+                        equal(expectedDictionaryValue),
+                        description: "Expected \(String(describing: dictionary["store"] as? String)) to become \(expectedDictionaryValue)."
+                    )
+                }
+                it("equals PLAY STORE when the store is play store") {
```

Comment:
All these tests are essentially the same with different inputs / outputs. Especially with Quick you could extract the core of the test then instantiate multiple tests (basically calls to `it`) inside of a loop to avoid the duplicate code.

Reply:
Good catch! I've updated the iOS & Android tests to be more compact.